### DISCUSSION TITLE
Printing the values of all tdms files in the same directory using the for loop in python

### DISCUSSION THREAD

**Discussion by adityabhandwalkar** [adamreeve/npTDMS#285] [Q&A]
I have a folder abc where I store my tdms files and I extract the data from those files using the the following function in python, now the problem is each time I run my main program I need to change the file number in the main so that I can run it, What I would like to do is that making changes into the load function and it will keep loop in same directory and print different values of files, lets say in abc I have file1.tdms, file2.tdms,... each time it should be printing values of different file in that folder abc

```
class loader():
    def load(self, path):

        
        tdms_file = TdmsFile.read(path)
        df = tdms_file.as_dataframe()
        file_read = TdmsFile.read(path)
        data = file_read.groups()[0].channels()[0].data
        
        plt.suptitle('Timeseries data', fontsize='30')
        plt.xlabel('Timestamp', fontsize ='20')
        plt.ylabel('feature values', fontsize='20')
        plt.plot(df.iloc[:,0])
        plt.show()
```

**Comment by Eifi1** [comment]
If i understand you correctly the problem is more system related than tdms related. You want to get all files from the directory abc right?
If I am right you can do this with this snippet to fetch all files ending with `.tdms`:

    from glob import iglob
    tdms_files = iglob("*.tdms")
    print(list(tdms_files)) # since the return of iglob is a generator

**Reply by adityabhandwalkar** [reply]
yes , But to be more specific What I want to do is that in the folder abc I have many files and using these following lines 
```
        file_read = TdmsFile.read(path)
        data = file_read.groups()[0].channels()[0].data
```
it writes a flat numpy array (a time series) , like `[0.312,0987,0.654,..]`  but this is from the `path `argument where I am providing address to the only one tdms file in the folder abc, what I want to do is that I would like to print the array of each and every file from the same folder and not the list of the file.
